src/redisearch_api.c

```diff
@@ -498,7 +498,7 @@ static RS_ApiIter* handleIterCommon(IndexSpec* sp, QueryInput* input, char** err
   RS_ApiIter* it = rm_calloc(1, sizeof(*it));
 
   if (input->qtype == QUERY_INPUT_STRING) {
-    if (QAST_Parse(&it->qast, &sctx, &options, input->u.s.qs, input->u.s.n, &status) !=
+    if (QAST_Parse(&it->qast, &sctx, &options, input->u.s.qs, input->u.s.n, RSGlobalConfig.defaultDialectVersion, &status) !=
```

Comment:
do we want to run this only on the default version?

Reply:
we can fix this after merging to the feature branch, for now, we got what we need